File: Week_08/Practicum/Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = dict()

        for i, num in enumerate(nums):
            d[num] = i

        for i, num in enumerate(nums):
            if target - num in d.keys() and i != d[target - num]:
                return [i, d[target - num]]

File: Week_08/Practicum/Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> values; // value -> index
        for (int i = 0; i < nums.size(); ++i) {
            if (values.count(target - nums[i])) {
                return { values[target - nums[i]], i};
            }
            values[nums[i]] = i;
        }

        return {};
    }
};
